### PR TITLE
[docs] update fingerprint example

### PR DESCRIPTION
**Pull request by jonsamp**
# Why

This updates the fingerprint example in the docs for workflows. This one should be copy/paste-able. The previous one was not.

# Screenshots

Screenshot 2024-11-20 at 12 02 52 PM

# Test Plan

Make sure the job is as simple as possible to show both local outputs and global outputs. Also make sure that this workflow works as expected.

### CODE REVIEW

File: docs/pages/workflows/variables.mdx [hunk 1]
```diff
@@ -12,36 +12,57 @@ jobs:
...
-        platform: ios
+  decide:
+    needs: [existing_build]
+    steps:
+      - name: decide
+        run: |
+          BUILD_ID="${{ needs.existing_build.outputs.build_id || '' }}"
+          echo $BUILD_ID
+
+          if [ -z "${BUILD_ID}" ] || [ "${BUILD_ID}" = "undefined" ]; then
```
Comment (sjchmiela) [inline docs/pages/workflows/variables.mdx:55]: I _think_ the `|| ''` added to the interpolation expression line above removes necessity to handle 'undefined' here.

**Review by sjchmiela — APPROVED** [review]
Thanks so much!!!

File: docs/pages/workflows/variables.mdx [hunk 2]
```diff
@@ -12,36 +12,57 @@ jobs:
...
+        run: |
+          BUILD_ID="${{ needs.existing_build.outputs.build_id || '' }}"
+          echo $BUILD_ID
+
+          if [ -z "${BUILD_ID}" ] ; then
+            echo "No build found. Should build."
+            exit 0
+          fi
+
+          echo "Found build: https://expo.dev/uuid/$BUILD_ID"
```
Comment (wschurman) [inline docs/pages/workflows/variables.mdx:60]: this endpoint isn't publicly-accessible

File: docs/pages/workflows/variables.mdx [hunk 3]
```diff
@@ -12,36 +12,57 @@ jobs:
...
+        id: fingerprint_hash_step
+        # @info Exposes the <CODE>fingerprint_hash_for_workflow</CODE> and <CODE>fingerprint_hash_for_steps</CODE> variables. #
+        outputs: [fingerprint_hash_for_workflow, fingerprint_hash_for_steps]
         # @end #
-        # @info Sets the <CODE>fingerprint_hash</CODE> as a variable #
+        # @info Sets the <CODE>fingerprint_hash_for_workflow</CODE> and <CODE>fingerprint_hash_for_steps</CODE> variables #
         run: |
-          FINGERPRINT_HASH=$(npx expo-updates fingerprint:generate --platform ios | jq '.hash')
-          set-output fingerprint_hash $FINGERPRINT_HASH
+          FINGERPRINT=$(npx expo-updates fingerprint:generate --platform ios)
```
Comment (wschurman) [inline docs/pages/workflows/variables.mdx:29]: this needs to do excludes based on workflow in order to generate consistent hashes across environments (before and after pod install for managed workflows for example)

**Review by wschurman — CHANGES_REQUESTED** [review]

**Pull request merged**